### ISSUE TITLE
New recommendations each time for same data

### ISSUE THREAD

**Issue by arundasK** [lyst/lightfm#355]
Hi, I am relatively new to recommendation systems and lightfm, library. I tried a tutorial program for music recommendation using implicit library. But every time I train model, I am getting different recommendations for same training data and user id. Please help me to understand what causes this issue. Thanks in advance.

**Comment by maciejkula** [comment]
The model uses randomness internally, so the results will be different with every run unless you fix the random seed. The construct of the `LightFM` class allows you to set the random state.

**Comment by maciejkula** [comment]
I'm uncomfortable with calling this a tunable hyperparameter. With the right (other) hyperparameters and enough data, the choice of random seed should have a small effect on the final accuracy. In fact, the setting is mostly there for testing purposes: in applications, you would not hold it fixed and let it very from fitting to fitting.

**Comment by arundasK** [comment]
@maciejkula  I highly appreciate your reply. Thanks for your time and effort.
